lambda/main.go

```diff
@@ -80,70 +65,97 @@ func Handler(ctx context.Context, evt json.RawMessage) (string, error) {
 		disableScaleOut, disableScaleIn bool
 	)
 
+	// Required environment variables
+	buildkiteQueue := os.Getenv("BUILDKITE_QUEUE")
+	if buildkiteQueue == "" {
+		return "", errors.New("BUILDKITE_QUEUE is required")
+	}
+
+	asgName := os.Getenv("ASG_NAME")
+	if asgName == "" {
+		return "", errors.New("ASG_NAME is required")
+	}
+
+	agentsPerInstanceStr := os.Getenv("AGENTS_PER_INSTANCE")
+	if agentsPerInstanceStr == "" {
+		return "", errors.New("AGENTS_PER_INSTANCE is required")
+	}
+	agentsPerInstance, err := strconv.Atoi(agentsPerInstanceStr)
+	if err != nil {
+		return "", fmt.Errorf("AGENTS_PER_INSTANCE must be an integer: %w", err)
+	}
```

Comment:
I would prefer a function for this

Reply:
I've refactored the environment variable loading back towards a few helper functions, which I feel also resolves the comments below. Since they all error out of the lambda handler, making it useless, I've leaned towards `log.Fatalf`.

Reply:
👍🏻